subsys/bluetooth/mesh/access.c

```diff
@@ -673,32 +717,24 @@ void bt_mesh_model_recv(struct bt_mesh_net_rx *rx, struct net_buf_simple *buf)
 		op->func(model, &rx->ctx, buf);
 		net_buf_simple_restore(buf, &state);
 	}
+
+#if defined(CONFIG_BT_MESH_ACCESS_LAYER_MSG)
+	if (msg_cb) {
```

Comment:
Wouldn't `if (IS_ENABLED(CONFIG_BT_MESH_ACCESS_LAYER_MSG) && msg_cb) { ...` make more sense here? The `msg_cb` variable isn't behind any #ifdef, so using IS_ENABLED means you won't generate an unused variable warning and the variable should still be optimised away from the final image latest at the linking stage.

Reply:
Yes, that would make more sense. This was accidentally left over from a previous change where `msg_cb` was behind an #ifdef. Since it no longer is I'll update this as recommended.